### ISSUE TITLE
Replace `SaveUpdateMixin.reload()` with django's `refresh_from_db` (?)

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#17838]
SaveUpdateMixin.reload() doesn't handle reverse relations for OneToOneFields - i.e. version.reload doesn't reload version.file - you have to specifically do version.file.reload(). We should look into dropping our special sauce and just use django's .refresh_from_db

**Comment by eviljeff** [comment]
More detail from working on a patch locally -  django's `refresh_from_db` uses the base_manager correctly and executes a query to get the new field values for the instance, (same as our `reload`, but a little more intelligently with respect to concrete and deferred fields) but then also also clears all the foreign key and related objects values (our `reload` doesn't do this).  This means every access of a relation results in a new query.  In a way, what we want - no cached values - but it also a large number of extra queries in a complex, interconnected object like `Addon`. In both cases transforms are executed when using the queryset from the base_manager, but the results are discarded because the transforms operate on the temporary instance that is only used to get updated values (i.e. wasted extra db queries).

Due to how our translations work, they aren't loaded lazily on accessing the translated field, so they're just missing entirely unless we manually load them after then `refresh_from_db`.

So, tl;dr: django's `refresh_from_db` doesn't really do a better job at reloading an object than our `reload`, it just does a better job at clearing cached values for related objects, so they are then loaded when accessed on demand, at the cost of extra db queries.

I'm not sure on the best way forward with this at the moment.  We ideally want a reload style function that uses the transforms, loads translations, and doesn't do any more queries than is necessary to refresh - neither django's or our special sauce do this.

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-56
